js/src/hooks/useIsMounted.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useCallback, useRef } from '@wordpress/element';
+
+/**
+ * Returns a function to check whether the caller component is mounted or unmounted.
+ * Usually, it's used to avoid the warning - Can't perform a React state update on an unmounted component.
+ *
+ * @return {() => boolean} A function returns a boolean indicates the caller component is mounted or unmounted.
+ */
+export default function useIsMounted() {
```

Comment:
:grey_question: 
Looks like native [`Node.isConnected`](https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected). I'm surprised that React does not have it somewhat built-in and we have to hand-craft it? Do we do something not popular/standard here?

Reply:
It had had a similar API `isMounted`, but was deprecated. https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html

`useIsMounted` is here mainly to avoid that scary warning appearing, but in fact there's no memory leak problem nor additional side effects.

The warning has quite a long background, mainly because some earlier global store/state management libraries based on the "Flux Architecture" had subscription and unsubscription mechanisms to be plugged in the class components. As an observer to get the global state changes, if a component forgets to unsubscribe after unmounted, it might let the global store keep the reference of the component itself or the method declared within the component, which may further lead to memory leaks.

However, with the implementation of something like react-redux, these subscription processes have been replaced by Higher-Order Components (HOC) and the memory leak problem is no longer common. After react hooks, these warnings are also almost false positives alarm. As it turns out, [this warning will be adjusted in the new version of react](https://github.com/facebook/react/pull/22114), and we will no longer need `useIsMounted` then.